core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -235,22 +289,31 @@ private boolean fullRefresh(MatViewRefreshTask refreshTask) {
                 engine.detachReader(baseTableReader);
                 refreshExecutionContext.of(baseTableReader);
                 try {
-                    final long toBaseTxn = baseTableReader.getSeqTxn();
-                    // Make time interval filter no-op as we're querying all partitions.
-                    refreshExecutionContext.setRange(Long.MIN_VALUE + 1, Long.MAX_VALUE);
-
                     walWriter.truncateSoft();
+
+                    final long toBaseTxn = baseTableReader.getSeqTxn();
                     final MatViewDefinition viewDef = state.getViewDefinition();
-                    insertAsSelect(state, viewDef, walWriter, toBaseTxn, refreshTriggeredTimestamp);
+                    // Specify -1 as the last refresh txn, so that we scan all partitions.
+                    final SampleByIntervalIterator intervalIterator = findSampleByIntervals(baseTableReader, viewDef, -1);
+                    if (intervalIterator != null) {
+                        insertAsSelect(state, viewDef, walWriter, intervalIterator, toBaseTxn, refreshTriggeredTimestamp);
+                    }
+
                     resetInvalidState(state, walWriter);
                 } finally {
                     refreshExecutionContext.clearReader();
                     engine.attachReader(baseTableReader);
                 }
+            } catch (Throwable th) {
+                LOG.error().$("full refresh error [view=").$(viewToken).$(", error=").$(th).I$();
+                refreshFailState(state, walWriter, microsecondClock.getTicks(), th.getMessage());
+                return false;
             }
         } catch (Throwable th) {
-            LOG.error().$("full refresh error [view=").$(viewToken).$(", error=").$(th).I$();
-            stateStore.removeViewState(viewToken);
```

Comment:
@eugenels removing the state doesn't seem to be the best approach here, so I swapped it with updating in-memory state only, without writing to WAL. This way we at least keep the in-memory state up-to-date.